src/tag-input/__tests__/TagInput.test.js

```diff
@@ -0,0 +1,126 @@
+    userEvent.type(screen.getByPlaceholderText(TEST_PLACEHOLDER), `${newTestVal}{enter}`)
+    expect(mockOnChange).toHaveBeenLastCalledWith(TEST_VALUES.concat([newTestVal]))
+    userEvent.type(screen.getByPlaceholderText(TEST_PLACEHOLDER), '{backspace}')
+    const valuesLastRemoved = TEST_VALUES.slice(0, -1)
+    expect(mockOnChange).toHaveBeenLastCalledWith(valuesLastRemoved)
+  })
+
+  it('tagSubmitKey', () => {
+    const mockOnAdd = jest.fn()
+    const newTestVal = 'Testing'
+    render(
+      <TagInput
+        tagSubmitKey="space"
+        values={TEST_VALUES}
+        onAdd={mockOnAdd}
+        inputProps={{ placeholder: TEST_PLACEHOLDER }}
+      />
+    )
+    userEvent.type(screen.getByPlaceholderText(TEST_PLACEHOLDER), `${newTestVal}{space}`)
+    expect(mockOnAdd).toHaveBeenCalledWith([newTestVal])
+  })
+
+  it('disabled', () => {
+    render(<TagInput disabled values={TEST_VALUES} inputProps={{ placeholder: TEST_PLACEHOLDER }} />)
+    TEST_VALUES.forEach(v => {
+      // Checks to make sure the "X" icon is not within each tag
+      expect(screen.getByText(v).children.length).toBe(0)
+    })
+    expect(screen.getByPlaceholderText(TEST_PLACEHOLDER)).toBeDisabled()
+  })
+
+  it('addOnBlur', () => {
+    const mockOnAdd = jest.fn()
+    jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => cb())
+    const newTestVal = 'Testing'
+    render(
+      <TagInput
+        addOnBlur
+        data-testid="wrapper"
+        values={TEST_VALUES}
+        onAdd={mockOnAdd}
+        inputProps={{ placeholder: TEST_PLACEHOLDER }}
+      />
+    )
+    userEvent.type(screen.getByPlaceholderText(TEST_PLACEHOLDER), `${newTestVal}`)
+    screen.getByPlaceholderText(TEST_PLACEHOLDER).blur()
+    fireEvent.blur(screen.getByPlaceholderText(TEST_PLACEHOLDER))
+    expect(mockOnAdd).toHaveBeenCalledWith([newTestVal])
+    window.requestAnimationFrame.mockRestore()
+  })
+
+  it('separator', () => {
+    const mockOnAdd = jest.fn()
+    const newTestVal = 'Testing|123'
+    render(
+      <TagInput separator="|" values={TEST_VALUES} onAdd={mockOnAdd} inputProps={{ placeholder: TEST_PLACEHOLDER }} />
+    )
+    userEvent.type(screen.getByPlaceholderText(TEST_PLACEHOLDER), `${newTestVal}{enter}`)
+    expect(mockOnAdd).toHaveBeenCalledWith(['Testing', '123'])
+  })
+
+  it('autocompleteItems', () => {
+    const mockOnAdd = jest.fn()
+    const testAutocompleteItems = ['Testing1', 'Testing2', 'Testing3', 'Other']
+    const testSearch = 'Test'
+    render(
+      <TagInput
+        autocompleteItems={testAutocompleteItems}
+        values={TEST_VALUES}
+        onAdd={mockOnAdd}
+        inputProps={{ placeholder: TEST_PLACEHOLDER }}
+      />
+    )
+    expect(screen.getByTestId('TagInput-autocomplete-toggle')).toBeInTheDocument()
+    testAutocompleteItems.forEach(i => {
+      expect(screen.queryByText(i)).not.toBeInTheDocument()
```

Comment:
Same comments as above for renaming `i` and breaking up the test with some whitespace. This one in particular is pretty complex - it looks like we're making sure the toggle button exists, the autocomplete menu items aren't visible until a value is typed, that the autocomplete menu items match the search string, and that clicking one of them runs the `onAdd` callback - that's a lot!

I would recommend breaking these up into separate tests, or at least adding comments above each act + assert phase to help the reader understand what it is we're testing